src/TiledArray/conversions/block_cyclic.h

```diff
@@ -44,8 +45,8 @@ namespace TiledArray {
 
 template <typename T,
           typename = scalapackpp::detail::enable_if_scalapack_supported_t<T>>
-class ScaLAPACKMatrix : public madness::WorldObject<ScaLAPACKMatrix<T>> {
-  using world_base_t = madness::WorldObject<ScaLAPACKMatrix<T>>;
+class BlockCyclicMatrix : public madness::WorldObject<BlockCyclicMatrix<T>> {
```

Comment:
should we namespace this so when Slate arrives we won't have to revisit?

Reply:
Do you mean in some sort of a `scalapack` namespace? I think that's what going on in the NWX PR viz linear algebra functionality (`heig`, etc). This strong type is just to manage block cyclic matrices in TA (a lot of other operations would need to be implemented to be of general use). SLATE has its own strong type (`slate::Matrix`), so we really just need to write conversion utilities:

```
BlockCyclicMatrix<T> array_to_block_cyclic
slate::Matrix<T> array_to_slate 
```

etc. But I separate this out into a separate namespace to make things more compartmentalized.

Reply:
These are namespaced, but I left the high-level wrappers (e.g. `heig`) as is to avoid clashes with the NWX PR

Reply:
@wavefunction91 you should feel free to cherry-pick the LAPACK stuff I did from NWX's TA and add it to this PR (or another PR). What I wrote isn't ready for prime time and being blunt getting it ready isn't one of my priorities so if we're waiting on me to do it ti could be awhile.

Reply:
TBH, I don't really have the cycles to cherry-pick and unit test your changes anytime in the near future either. As this PR is ScaLAPACK oriented, I think its fine if we leave that off onto a separate PR if @evaleev is fine with that